client/src/containers/SignUp/SignUp.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom'
+import SmallContainer from '../../components/SmallContainer/SmallContainer';
+import InputField from '../../components/InputField/InputField';
+import { getError, hasError, isTouched } from '../../libs/validate';
+import handleErrors from './Schema';
+import './SignUp.css';
+
+const SignUp = () => {
+  const [ user, setUser ] = useState({ name: '', email: '', password: '' });
+  const [ touched, setTouched ]= useState({});
+  const [ error, setError ] = useState({});
+  
+  const touchedFields = {}
+  
+  const handleChange = field => event => {
+    setUser({ ...user, [field]: event.target.value }, () => handleErrors(user, setError));
+  }
+
+  const handleBlur = (field) => (event) => {
+    handleErrors(user, setError);
+    touchedFields[field] = true;
+    setTouched({ ...touched, ...touchedFields})
+  }
+
+  return (
+    <SmallContainer>
+      <form className="has-text-centered">
+        <figure className="image is-128x128">
+          <img src="https://dummyimage.com/100x100/000/fff&text=CoderHood" alt="CoderHood" className="brand-logo is-rounded"/>
+        </figure>
+
+        <InputField 
```

Comment:
just create a renderInput function which can take the input type (name, email, pwd) as argument. And return this input field. Try to remove duplicates as much as possible. Code Climate is also warning about the duplicates.

Reply:
Okay, I'll do.